pkg/prove/proof.go

```diff
@@ -107,6 +117,49 @@ func txSharePosition(txs types.Txs, txIndex uint64) (startSharePos, endSharePos
 	return startSharePos, endSharePos, nil
 }
 
+// MsgSharesPosition returns the start and end positions for the shares
+// where a given message, referenced by its wrapped pfb transaction, was published at.
+// Note: only supports transactions containing a single message
+func MsgSharesPosition(tx types.Tx) (beginShare uint64, endShare uint64, err error) {
+	unwrappedTx, isMalleated := types.UnmarshalIndexWrapper(tx)
+	if !isMalleated {
+		return beginShare, endShare, fmt.Errorf("not a malleated tx")
+	}
+
+	encCfg := encoding.MakeConfig(blobmodule.AppModuleBasic{})
+	decoder := encCfg.TxConfig.TxDecoder()
+
+	decodedTx, err := decoder(unwrappedTx.Tx)
+	if err != nil {
+		return beginShare, endShare, err
+	}
+
+	if len(decodedTx.GetMsgs()) == 0 {
+		return beginShare, endShare, fmt.Errorf("pfb contains no messages")
```

Comment:
[nit][proposal] to explicitly spell out PFB in error messages because they may be seen and reported by users who are not aware of Celestia specific acronyms.

```suggestion
		return beginShare, endShare, fmt.Errorf("PayForBlob contains no messages")
```

Reply:
4595dd2